addons/account/static/src/js/reconciliation/reconciliation_model.js

```diff
@@ -972,47 +966,45 @@ var StatementModel = BasicModel.extend({
             var line = _.find(self.lines, function (l) {
                 return l.id === data.st_line.id;
             });
+            // Line has been already reconciled by the reconciliation model.
+            if(data.status === 'reconciled'){
+                self.valuenow += 1;
+                line.reconciled = true;
+                line.reconciled_aml_ids = data.reconciled_aml_ids;
+                return;
+            }
             line.visible = true;
             line.limitMoveLines = self.limitMoveLines;
             _.extend(line, data);
             self._formatLineProposition(line, line.reconciliation_proposition);
             if (!line.reconciliation_proposition.length) {
                 delete line.reconciliation_proposition;
-            }else{
-                // loop state propositions
-                var debit_props = _.filter(line.reconciliation_proposition, function(p){
-                    return p.amount > 0 && line.st_line.amount;
-                });
-                var credit_props = _.filter(line.reconciliation_proposition, function(p){
-                    return p.amount < 0 && line.st_line.amount < 0;
-                });
-                var sum_debit_props = debit_props.length > 0 ? debit_props.map(function(p){return p.amount}).reduce(function(p1, p2){return p1 + p2}) : 0;
-                var sum_credit_props = credit_props.length > 0 ? credit_props.map(function(p){return p.amount}).reduce(function(p1, p2){return p1 + p2}) : 0;
-
-                // Want to display the triangle for partial reconciliation or not.
-                if(line.st_line.amount_currency > 0 && line.st_line.amount_currency < sum_debit_props && debit_props.length == 1)
-                    debit_props[0].display_triangle = true;
-                else if(line.st_line.amount_currency < 0 && line.st_line.amount_currency > sum_credit_props && credit_props.length == 1)
-                    credit_props[0].display_triangle = true;
             }
 
             // No partner set on st_line and all matching amls have the same one: set it on the st_line.
-            if(!line.st_line.partner_id && line.reconciliation_proposition){
-                var hasDifferentPartners = function(prop){
-                    return !prop.partner_id || prop.partner_id != line.reconciliation_proposition[0].partner_id;
-                }
-
-                if(!_.any(line.reconciliation_proposition, hasDifferentPartners)){
-                    defs.push(self.changePartner(line.handle, {
-                        'id': line.reconciliation_proposition[0].partner_id,
-                        'display_name': line.reconciliation_proposition[0].partner_name,
-                    }, true));
-                }
-            }
+            defs.push(
+                $.when(self._computeLine(line))
```

Comment:
actually the $when(self._computeLine(line)) is not necessary since computeLine is already a deferred and could be rewritten `defs.push(
    self._computeLine(line)
    .then(function(){})
        .then(function(){})
)`
`return $.when(defs)`